src/main/resources/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.R

```diff
@@ -0,0 +1,118 @@
+yieldData        = read.table(args[3], sep="\t", header=T)
+umiData          = read.table(args[4], sep="\t", header=T)
+outputFile       = args[5]
+sampleInfo       = paste(args[6:length(args)])
+
+pdf(outputFile, width=11, height=8.5)
+
+# Plot #1 - Family Sizes for different kinds of familes
+ggplot(familyData) + aes(x=family_size) +
+  geom_line(aes(y=ds_count, color="DS Families")) +
+  geom_line(aes(y=ss_count, color="SS Families")) + 
+  geom_line(aes(y=cs_count, color="By Position")) + 
+  scale_x_continuous(trans="log2", minor_breaks=seq(0,max(familyData$family_size), by=2)) +
+  scale_color_manual(values=fgcolors) +
+  labs(x="Family Size (log2 scaled)", y="Count of Families", title=paste("Family Size Distributions for", sampleInfo)) +
+  theme(plot.title = element_text(hjust = 0.5), legend.title=element_blank())
+
+
+# Plot #2- Cumulative Family Sizes for different kinds of familes
+ggplot(familyData) + aes(x=family_size) +
+  geom_line(aes(y=cs_fraction_gt_or_eq_size, color="By Position"), alpha=0.5) + 
+  geom_line(aes(y=ss_fraction_gt_or_eq_size, color="SS Families"), alpha=0.5) + 
+  geom_line(aes(y=ds_fraction_gt_or_eq_size, color="DS Families"), alpha=0.5) +
+  scale_x_continuous(trans="log2", minor_breaks=seq(0,max(familyData$family_size), by=2)) +
+  scale_color_manual(values=fgcolors) +
+  labs(x="Family Size (log2 scaled)", y="Fraction of Families at >= Family Size", title=paste("Cumulative Family Size Distributions for", sampleInfo)) +
+  theme(plot.title = element_text(hjust = 0.5), legend.title=element_blank())
+
+# Plot #3a - Duplex Family Size Distribution
+ggplot(duplexFamilyData) +
+  aes(x=ab_size, y=ba_size, z=count) +
+  stat_summary_2d(fun=sum, binwidth=1) +
+  scale_fill_gradient2(low=fgblue, mid="white", high=fggreen, space="Lab",
+                       midpoint=log10(max(duplexFamilyData$count)), guide = "colourbar", trans="log2") +
+  labs(x="AB Reads", y="BA Reads", title=paste("Duplex Tag Family Size Distribution for", sampleInfo)) +
+  theme(plot.title = element_text(hjust = 0.5))
+
+# Plot #3b - Duplex Family Size Distribution for families with ab > 1 and ba > 1
+ggplot(subset(duplexFamilyData, duplexFamilyData$ba_size > 0)) +
+  aes(x=ab_size, y=ba_size, z=count) +
+  stat_summary_2d(fun=sum, binwidth=1) +
+  scale_fill_gradient2(low=fgblue, mid="white", high=fggreen, space="Lab",
+                       midpoint=log10(max(duplexFamilyData$count)), guide = "colourbar", trans="log2") +
+  labs(x="AB Reads", y="BA Reads", title=paste("Duplex Tag Family Size Distribution for ", sampleInfo, " ; Only Families with AB>0 & BA>0", sep="")) +
+  theme(plot.title = element_text(hjust = 0.5))
+
+# Plot #4 - Duplex Yield
+ggplot(yieldData) +
+  aes(x=read_pairs) +
+  geom_area(aes(y=ds_families, fill="DS Families")) +
+  geom_area(aes(y=ds_families * ds_fraction_duplexes_ideal, fill="Duplexes - Ideal")) +
+  geom_area(aes(y=ds_duplexes, fill="Duplexes - Actual")) +
```

Comment:
That's why it's _after_.  Since these are distinct `geom_areas` they are added to the plot in the order they are specified here.  There are always more families than ideal duplexes, and more ideal duplexes than actual duplexes, so doing it this way they stack nicely and are all visibible.  Re-arranging would mean that those with more would hide those with less.